### ISSUE TITLE
Use array compression to save data

### ISSUE THREAD

**Issue by elcorto** [mala-project/mala#436]
Consider using a library that can do compression, such as [zarr](https://github.com/zarr-developers/zarr-python), HDF5 or openPMD. `zarr` uses [blosc](https://www.blosc.org/python-blosc/python-blosc.html) by default for compression. Plain `blosc` has a size limit of 2 GiB and requires [manual chunking](https://www.blosc.org/python-blosc/tutorial.html#compressing-from-a-data-pointer) beyond that point, whereas `zarr` does automatic chunking.

```py
# Create 7.5 GiB of data
>>> x=linspace(0, 100, int(1e9))
>>> x.nbytes / 1024**3
7.450580596923828

>>> xz=zarr.array(x)
>>> xz.info
Type               : zarr.core.Array
Data type          : float64
Shape              : (1000000000,)
Chunk shape        : (488282,)
Order              : C
Read-only          : False
Compressor         : Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)
Store type         : zarr.storage.KVStore
No. bytes          : 8000000000 (7.5G)
No. bytes stored   : 294795139 (281.1M)
Storage ratio      : 27.1
Chunks initialized : 2048/2048

>>> zarr.save("xz.zarr", xz)

# Check "No. bytes stored" on disk
>>> !du -hs xz.zarr
285M    xz.zarr
```

Compression rate of 27, not bad.

**Comment by elcorto** [comment]
Might be duplicate of / superseded by #410 #417

**Comment by RandomDefaultUser** [comment]
Yes, that is a good point. The OpenPMD support is mostly ready to go, and will be part of the next release. I think this issue can be closed once #417 is merged.

**Comment by RandomDefaultUser** [comment]
Closed because #417 is merged now.
